src/Common/tests/Collections/ICollectionTest.cs

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Linq;
+using Xunit;
+using Xunit.Sdk;
+// ReSharper disable InconsistentNaming
+
+namespace Tests.Collections
+{
+    public abstract class ICollectionTest<T> : IEnumerableTest<T>
+    {
+        internal class MyInvalidReferenceType
+        {
+        }
+
+        internal struct MyInvalidValueType
+        {
+        }
+
+        protected ICollection GetCollection(object[] items)
+        {
+            IEnumerable obj = GetEnumerable(items);
+            Assert.IsAssignableFrom<ICollection>(obj);
+            return (ICollection)obj;
+        }
+        protected Type[] ValidArrayTypes { get; set; }
+
+        protected Type[] InvalidArrayTypes { get; set; }
+
+        protected abstract bool ItemsMustBeUnique { get; }
+        protected abstract bool ItemsMustBeNonNull { get; }
+        protected bool ExpectedIsSynchronized { get; private set; }
+        protected virtual bool CopyToOnlySupportsZeroLowerBounds
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        protected ICollectionTest(bool isSynchronized)
+        {
+            ExpectedIsSynchronized = isSynchronized;
+            ValidArrayTypes = new[] { typeof(object) };
+            InvalidArrayTypes = new[]
+            {
+                typeof (MyInvalidReferenceType),
+                typeof (MyInvalidValueType)
+            };
+        }
+
+        [Fact]
+        public void Count()
+        {
+            var items = GenerateItems(16);
```

Comment:
You have a lot of "var"s in here that violate the style guide (https://github.com/dotnet/corefx/wiki/Coding-style).  Notably "We only use var when it's obvious what the variable type is (i.e. var stream = new FileStream(...) not var stream = OpenStandardInput())."